src/autopas/containers/cellPairTraversals/CSlicedBasedTraversal.h

```diff
@@ -0,0 +1,235 @@
+   * allCells=false, if you plan to just iterate over all cells, e.g., to iterate over verlet lists saved within the
+   * cells, use allCells=true. For the sliced step if allCells is false, iteration will not occur over the last layer of
+   * cells (for _overlap=1) (in x, y and z direction).
+   */
+  template <bool allCells = false, typename LoopBody>
+  inline void cSlicedTraversal(LoopBody &&loopBody);
+
+  /**
+   * Overlap of interacting cells. Array allows asymmetric cell sizes.
+   */
+  std::array<unsigned long, 3> _overlap;
+
+ private:
+  /**
+   * Store ids of dimensions ordered by number of cells per dimensions.
+   */
+  std::array<int, 3> _dimsPerLength;
+
+  /**
+   * Interaction length (cutoff + skin).
+   */
+  double _interactionLength;
+
+  /**
+   * Cell length in CellBlock3D.
+   */
+  std::array<double, 3> _cellLength;
+
+  /**
+   * Overlap of interacting cells along the longest axis.
+   */
+  unsigned long _overlapLongestAxis;
+
+  /**
+   * The number of cells per slice in the dimension that was sliced.
+   */
+  std::vector<unsigned long> _sliceThickness;
+
+  /**
+   * Data Layout Converter to be used with this traversal.
+   */
+  utils::DataLayoutConverter<PairwiseFunctor, dataLayout> _dataLayoutConverter;
+};
+
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3>
+inline void CSlicedBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3>::init(
+    const std::array<unsigned long, 3> &dims) {
+  for (unsigned int d = 0; d < 3; d++) {
+    _overlap[d] = std::ceil(_interactionLength / _cellLength[d]);
+  }
+
+  // find longest dimension
+  auto minMaxElem = std::minmax_element(this->_cellsPerDimension.begin(), this->_cellsPerDimension.end());
+  _dimsPerLength[0] = (int)std::distance(this->_cellsPerDimension.begin(), minMaxElem.second);
+  _dimsPerLength[2] = (int)std::distance(this->_cellsPerDimension.begin(), minMaxElem.first);
+  _dimsPerLength[1] = 3 - (_dimsPerLength[0] + _dimsPerLength[2]);
+
+  _overlapLongestAxis = _overlap[_dimsPerLength[0]];
+}
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3>
+template <bool allCells, typename LoopBody>
+void CSlicedBasedTraversal<ParticleCell, PairwiseFunctor, dataLayout, useNewton3>::cSlicedTraversal(
+    LoopBody &&loopBody) {
+  using std::array;
+
+  auto numSlices = _sliceThickness.size();
+  // 0) check if applicable
+
+  std::array<size_t, 2> overLapps23{_overlap[_dimsPerLength[1]], _overlap[_dimsPerLength[2]]};
+  if (allCells) {
+    overLapps23 = {0ul, 0ul};
+    _sliceThickness.back() += _overlapLongestAxis;
+  }
+
+  for (size_t offset = 0; offset < 2; offset++) {
+#ifdef AUTOPAS_OPENMP
+// although every thread gets exactly one iteration (=slice) this is faster than a normal parallel region
+#pragma omp parallel for schedule(dynamic, 1)
```

Comment:
wouldn't a static scheduling make more sense or is there no performance difference?